Mailbox.Tests/DataLoaderTests.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace Mailbox.Tests
+{
+    [TestClass]
+    public class DataLoaderTests
+    {
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void DataLoader_Load_StreamIsNull_ThrowNullExeption()
+        {
+            DataLoader dL = new DataLoader(null);
+        }
+
+        [TestMethod]
+        public void DataLoader_Load_NotValidJson_ReturnsNull()
+        {
+            string filePath = Path.GetRandomFileName();
+            Stream stream = GenerateTextFile(filePath);
+            DataLoader dL = new DataLoader(stream);
+            List<Mailbox> worked = dL.Load();
+
+            stream.Close();
+            File.Delete(filePath);
+            Assert.IsNull(worked);
+        }
+
+        [TestMethod]
+        public void DataLoader_Load()
+        {
+            string filePath = Path.GetRandomFileName();
+            Stream stream = GenerateJsonFile(filePath);
```

Comment:
I believe we are supposed to use a [MemoryStream](https://docs.microsoft.com/en-us/dotnet/api/system.io.memorystream?view=netframework-4.8) to unit test the IO portion of our load and save methods
```suggestion
            MemoryStream memoryStream = new MemoryStream();
```

Reply:
Thank you! Totally forgot when I was writing my tests.